python/src/equistore/operations/to.py

```diff
@@ -0,0 +1,179 @@
+    :param requires_grad: :py:class:`bool`, whether or not to use torch's
+        autograd to record operations on this tensor. Only applicable if
+        `backend` is set to ``"torch"``.
+    :param dtype: :py:class:`torch.dtype`, the base dtype of the resulting torch
+        tensor, only applicable if `backend` is set to ``"torch"``.
+    :param device: :py:class:`torch.device`, the device on which the tensors of
+        the resulting :py:class:`TensorMap` should be stored, only applicable if
+        `backend` is set to ``"torch"``.
+
+    :return: a :py:class:`TensorMap` or :py:class:`TensorBlock` converted to the
+        specified backend.
+    """
+    if not isinstance(tensor, TensorMap):
+        raise TypeError(
+            f"``tensor`` should be an equistore `TensorMap`, got {type(tensor)}"
+        )
+    
+    keys = tensor.keys
+    torch_blocks = [
+        block_to(
+            tensor[key].copy(), 
+            backend=backend,
+            requires_grad=requires_grad, 
+            dtype=dtype, 
+            device=device
+        )
+        for key in keys
+    ]
+    
+    return TensorMap(keys=keys, blocks=torch_blocks)
+
+
+def block_to(
+    block: TensorBlock,
+    backend: str,
+    requires_grad: Optional[bool] = None,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+) -> TensorBlock:
+    """
+    Converts a :py:class:`TensorMap` or :py:class:`TensorBlock` to a different
+    backend. Currently only supports converting to numpy or torch-based tensors.
+
+    :param tensor: input :py:class:`TensorMap` or :py:class:`TensorBlock`.
+    :param backend: :py:class:`str`, the backend to convert to. Currently only
+        supports ``"numpy"`` or ``"torch"``.
+    :param requires_grad: :py:class:`bool`, whether or not to use torch's
+        autograd to record operations on this tensor. Only applicable if
+        `backend` is set to ``"torch"``.
+    :param dtype: :py:class:`torch.dtype`, the base dtype of the resulting torch
+        tensor, only applicable if `backend` is set to ``"torch"``.
+    :param device: :py:class:`torch.device`, the device on which the tensors of
+        the resulting :py:class:`TensorMap` should be stored, only applicable if
+        `backend` is set to ``"torch"``.
+
+    :return: a :py:class:`TensorMap` or :py:class:`TensorBlock` converted to the
+        specified backend.
+    """
+    if not isinstance(block, TensorBlock):
+        raise TypeError(
+            f"``block`` should be an equistore `TensorBlock`, got {type(block)}"
+        )
+    if not isinstance(backend, str):
+        raise TypeError("`backend` should be passed as a `str`")
+    if backend == "torch":
+        # Check args
+        if not isinstance(requires_grad, bool):
+            raise TypeError("`requires_grad` should be passed as a `bool`")
+        if not isinstance(dtype, torch.dtype):
+            raise TypeError("`dtype` should be passed as a `torch.dtype`")
```

Comment:
Not really, torch also accepts strings here. Overall, I think it is fine not to type check these, and pass them through to the backend which will handle them for us.